src/controller/CHIPDeviceController.cpp

```diff
@@ -3037,25 +3070,36 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
     }
     case CommissioningStage::kAttestationVerification: {
         ChipLogProgress(Controller, "Verifying attestation");
-        if (!params.GetAttestationElements().HasValue() || !params.GetAttestationSignature().HasValue() ||
-            !params.GetAttestationNonce().HasValue() || !params.GetDAC().HasValue() || !params.GetPAI().HasValue() ||
-            !params.GetRemoteVendorId().HasValue() || !params.GetRemoteProductId().HasValue())
+        VerifyOrReturn(IsAttestationInformationMissing(params) == false);
```

Comment:
OK, so this is _really_ misleading.  It looks like a boolean check, but it has various side-effects.  I would suggest naming it like so:
```suggestion
        VerifyOrReturn(!CompleteCommissioningStageDueToMissingAttestionInformation(params));
```
or something.  Or just having it be a true boolean check with no side-effects (static method, also), and doing `CommissioningStageComplete` in its callers.

Reply:
Fixed in latest commit.